### DISCUSSION TITLE
Is there any way to better display/visualize binary values of busses, registers, ... ?

### DISCUSSION THREAD

**Discussion by LAPSyLAB** [logisim-evolution/logisim-evolution#1751] [General]
Hello,

we really like this tool and we make CPU models for use in teaching process at our university.
I always wanted to ask community, whether there is an element or any other way to display decoded text information based 
on binary value in register, bus or any other multibit connection or element.

For instance, we would like to display different texts for values of 4 bits (0-15) - for instance instruction names in text.

Any way to do this already or how to approach adding this feature, if it is not yet available.

Thanks in advance,
regards.

**Comment by LAPSyLAB** [comment]
Hello.

we're using Evolution for teaching (have working CPU models) and would really like to have this feature, so we can display texts according to values... 

Any hint, where we could start adding this feature ?

Thanks.

**Reply by maehne** [reply]
Maybe, this could be implemented as a new Java component, which could be connected to any bus and which offers properties to specify for what bit pattern what output should be displayed. You could have a look to how the "Wiring/Probe" component is [implemented](https://github.com/logisim-evolution/logisim-evolution/blob/main/src/main/java/com/cburch/logisim/std/wiring/Probe.java). Maybe, the latter could be also extended with a "Custom" Radix, which if selected allows, e.g., specifying an alias file (inspired from [GTKwave](https://gtkwave.sourceforge.net/) manual section "Alias Files and Attaching External Disassemblers") or entering an alias table.
